### ISSUE TITLE
Unselectable file/folder confusion in column view

### ISSUE THREAD

**Issue by wout** [elementary/files#767]
I'm experiencing some weird behaviour in column view. Once a folder is opened, only the top level content is editable.

For example, here everything is fine. I can right click on any item and modify them using right click:

![first-column-selectable](https://user-images.githubusercontent.com/107324/48659038-bd0d1c80-ea4b-11e8-8727-1fdf4305dfb1.png)

Once I am inside a folder, only the highest level can be edited. There is no way to edit the name of items in the first column:

![first-column-unselectable](https://user-images.githubusercontent.com/107324/48659046-de6e0880-ea4b-11e8-8bea-705cd71e1808.png)

What's more, if I right click on any directory, it's selected visually, but the this does not change the state of the left column:

![first-column-right-click-confusion](https://user-images.githubusercontent.com/107324/48659060-1aa16900-ea4c-11e8-8b96-9a7bb330ed9f.png)

So now we are looking at the contents of the "appcenter" folder, while the "mail" folder is selected.

**Comment by jeremypw** [comment]
There is more than one issue here, although possibly related.

1)  Renaming in higher level folders.  This is possible using the keyboard.  Use the arrow keys to move the cursor and select a folder in a higher level folder  (Left arrow to move to higher level and up/down arrow to select a folder).  Press F2 to rename the folder with keyboard focus.  On current master. right-clicking on a folder in a higher level folder and using the context menu also works even if that folder is not initially focused.

2) I cannot confirm the second issue - clicking on a folder both opens it at the right and also moves the cursor to that folder.  It is however possible to use the keyboard to move the cursor without opening a folder to the right - this is intended.

The above is using the default "single-click to activate" mode.  Have you changed this setting to the double-click mode?

**Comment by wout** [comment]
In the meantime I reinstalled Pop!_OS (where I came from), so I can't provide more information. There are so many problems with elementary OS that I couldn't deal with it any longer. It's a shame, because it's beautiful, but Juno feels like an early beta release in its current state.

**Comment by jeremypw** [comment]
OK, thanks for taking the trouble to comment anyway.  I hope you give elementary a try later.

**Comment by Akryum** [comment]
I can reproduce the second issue: right-clicking to another folder name change its state to selected but doesn't switch the child column content. Also, another issue (maybe related): the folder icon stays in 'folder open' state even when selecting other folders, re-selecting the folder, etc.

**Comment by jeremypw** [comment]
You are right that right-clicking a folder does not activate it (by design)
but it does leave it selected if you cancel the context menu.  This means
you can have a folder selected but not expanded into the child column.  But
I do not think this is a bug - you can have the same situation in list view
and using other file-managers.  It is the primary button that is intended
to activate (and expand) folders.

**Comment by jeremypw** [comment]
Closing as the issue appears incomplete and multifold.  Feel free to open separate well-defined issue reports relating to the latest version of Files.
